source/elements/oneDAL/source/algorithms/decomposition/pca.rst

```diff
@@ -1,3 +1,447 @@
+.. highlight:: cpp
+.. default-domain:: cpp
+
 ===================================
 Principal Components Analysis (PCA)
 ===================================
+
+Principal Component Analysis (PCA) is an algorithm for exploratory data analysis
+and dimensionality reduction. PCA transforms a set of observations of possibly
+correlated variables to a new set of uncorrelated variables, called principal
+components. Principal components are the directions of the largest variance,
+that is, the directions where the data is mostly spread out.
+
+Given the training set :math:`X = \{ x_1, \ldots, x_n \}` of
+:math:`p`-dimensional samples and the number of principal components :math:`r`,
+the problem is to compute :math:`r` principal directions (:math:`p`-dimensional
+eigenvectors) for the training set. The eigenvectors can be grouped into the
+:math:`r \times p` matrix :math:`T` that contains one eigenvectors in each row.
+
+-----------------------
+Covariance-based method
+-----------------------
+[TBD]
+
+----------------
+SVD-based method
+----------------
+[TBD]
+
+-------------------
+Sign-flip technique
+-------------------
+Eigenvectors computed by some eigenvalue solvers are not uniquely defined due to
+sign ambiguity. To get the deterministic result, a sign-flip technique should be
+applied. One of the sign-flip techniques proposed in [Bro07]_ requires the
+following modification of eigenvectors matrix :math:`T`,
+
+.. math::
+   \hat{T}_i = T_i \cdot \mathrm{sgn}(\max_{1 \leq j \leq p } |{T}_{ij}|), \quad 1 \leq i \leq r,
+
+where :math:`T` is eigenvectors matrix, :math:`T_i` is :math:`i`-th row,
+:math:`T_{ij}` is the element in the :math:`i`-th and :math:`j`-th column,
+:math:`\mathrm{sgn}(\cdot)` is signum function,
+
+.. math::
+   \mathrm{sgn}(x) =
+      \begin{cases}
+         -1, & x < 0, \\
+          0, & x = 0, \\
+          1, & x > 0.
+      \end{cases}
+
+.. note::
+   The sign-flip technique described above is provided as an example. oneDAL
+   spec does not require implementation of this sign-flip technique.
+   Implementers can choose an arbitrary technique that modifies sign of the
+   eigenvectors or use deterministic algorithm for eigenvectors computations.
+
+
+-------------
+Usage example
+-------------
+::
+
+   onedal::pca::model run_training(const onedal::table& data) {
+
+      const auto pca_desc = onedal::pca::desc<float>{}
+         .set_component_count(5)
+         .set_is_deterministic(true);
+
+      const auto result = onedal::train(pca_desc, data);
+
+      print_table("means", result.get_means());
```

Comment:
as we use this print_table function in multiple examples we would need to explain somewhere what it is